src/Core/Settings.cpp

```diff
@@ -5475,6 +5475,9 @@ Allow to ignore schema evolution in Iceberg table engine and read all data using
 :::note
 Enabling this setting can lead to incorrect result as in case of evolved schema all data files will be read using the same schema.
 :::
+)", 0) \
+    DECLARE(Int64, iceberg_query_at_timestamp_ms, 0, R"(
```

Comment:
IMO it would be better to create a new setting field type (add class `SettingFieldDateTime` into [`SettingField.h`](https://github.com/ClickHouse/ClickHouse/blob/master/src/Core/SettingsFields.h) ). With the new `SettingFieldDateTime` you have complete control over parsed of setting value from string, see `SettingFieldNumber` for some inspiration.


This line would be then changed to:
```suggestion
    DECLARE(DateTime, iceberg_query_at_timestamp_ms, 0, R"(
```

Reply:
Thanks!

Reply:
I checked this out and I'd like to do this but might need some more guidance on navigating the metaprogramming going on here. 🫠 

I'll try again next week.